crypto/bio/bss_bio.c

```diff
@@ -119,16 +119,13 @@ struct bio_bio_st {
 
 static int bio_new(BIO *bio)
 {
-    struct bio_bio_st *b;
+    struct bio_bio_st *b = OPENSSL_zalloc(sizeof(*b));
 
-    b = OPENSSL_malloc(sizeof(*b));
     if (b == NULL)
         return 0;
 
-    b->peer = NULL;
```

Comment:
No.  The null pointer isn't necessarily all 0 bits.  See http://c-faq.com/null/runtime0.html